web/modules/custom/profile_block/src/Plugin/Block/user_profile_block.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Drupal\profile_block\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use \Drupal\node\Entity\Node; 
+
+/** 
+ *
+ * @Block(
+ *   id = "user_profile_block",
+ *   admin_label = @Translation("user profile block"),
+ *   category = @Translation("custom World"),
+ * )
+ */
+class user_profile_block extends BlockBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $db = \Drupal::database(); 
+
+      //for username
+      $user = \Drupal::routeMatch()->getParameter('user');
+      $uid = $user->id();
+ 
+      $query1 = $db->select('users_field_data', 'e');
+      $query1->condition('e.uid', $uid );
+      $query1->fields('e'); 
+      $usrn = $query1->execute()->fetchAll(); 
+
+      foreach($usrn as $row){
+        $uname = $row->name;
+      }
+
+      //for post count
+      $post_count = 0;
+
+      $query2 = $db->select('node_field_data', 'n');
+      $query2->condition('n.uid', $uid );
+      $query2->condition('n.status', 1 );
+      $query2->fields('n'); 
+      $posts = $query2->execute()->fetchAll();
+
+      foreach($posts as $row){
+        $post_count++;
+      } 
+
+      //for followers count
+      $query3 = $db->select('flag_counts', 'f');
+      $query3->condition('f.entity_id', $uid );
+      $query3->fields('f'); 
+      $followers = $query3->execute()->fetchAll();
+
+      $followers_count = 0;
+      foreach($followers as $row){
+        $followers_count = $row->count;
+      }  
+
+      //for following count
+      $following_count = 0;
+
+      $query4 = $db->select('flagging', 'fl');
+      $query4->condition('fl.uid', $uid );
+      $query4->condition('fl.flag_id', 'following' );
+      $query4->fields('fl'); 
+      $following = $query4->execute()->fetchAll();
+
+      foreach($following as $row){
+        $following_count++;
+      } 
+
+      //for bio
+      $query5 = $db->select('user__field_bio', 'b');
+      $query5->condition('b.entity_id', $uid );
+      $query5->fields('b'); 
+      $result5 = $query5->execute()->fetchAll(); 
+
+      foreach($result5 as $row){
+        $bio = $row->field_bio_value;  
+      } 
+
+      //for full name
+      $query6 = $db->select('user__field_full_name', 'fn');
```

Comment:
Use entity type manager and user entity to get the data, instead of direct sql query

Reply:
changes done here : 
https://github.com/guptahemant/insta-clone/pull/87/files#diff-7b04f370db26fe83d27b4f6a39974f2735d982f7fd33c9aa6c72c75a91aeeee1